framework/auth/core.py

```diff
@@ -541,14 +582,35 @@ def _set_email_token_expiration(self, token, expiration=None):
         self.email_verifications[token]['expiration'] = expiration
         return expiration
 
-    def add_email_verification(self, email, expiration=None):
+    def add_unconfirmed_email(self, email, expiration=None):
         """Add an email verification token for a given email."""
+        # TODO: If the unconfirmed email is already on the User, refresh the token
+
+        # verify email
+        if not re.match(r'^.+@[^.].*\.[a-z]{2,10}$', email, re.IGNORECASE):
+            raise ValueError("Invalid Email")
+
         token = generate_confirm_token()
 
         self.email_verifications[token] = {'email': email.lower()}
         self._set_email_token_expiration(token, expiration=expiration)
         return token
 
+    def remove_unconfirmed_emails(self, *args):
```

Comment:
True, but then you couldn't the keyword arguments in the method definition. You'd have to do:

```python
def remove_unconfirmed_emails(self, save=False , *args):

# or

def remove_unconfirmed_emails(self, *args, **kwargs):
    save = kwargs.get('save', False)
```

Neither is ideal.